src/main/java/ch/baloise/corellia/api/entities/Cancellation.java

```diff
@@ -1,52 +1,47 @@
 package ch.baloise.corellia.api.entities;
 
-import static ch.baloise.corellia.api.constraints.SizeConstraint.CONTRACT_ID_MAX_SIZE;
-
 import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 import java.io.Serializable;
-import java.time.LocalDate;
+import javax.validation.Valid;
 import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
 
 public class Cancellation implements Serializable {
 
-  private static final long serialVersionUID = 10;
+    private static final long serialVersionUID = 10;
 
-  @NotNull
-  @Size(max = CONTRACT_ID_MAX_SIZE)
-  @JsonPropertyDescription(
-      "Id given by SaaS provider. Identifies the contract which should be cancelled")
-  private String contractId;
+    @NotNull
+    @Valid
+    private Identifier identifier;
 
-  @NotNull
-  @JsonPropertyDescription("effecitve date of cancellation")
-  private LocalDate effectiveDate;
+    @NotNull
+    @Valid
+    private Action action;
 
-  @NotNull
-  @JsonPropertyDescription("A code uniquely identifying the cancellation reason")
-  private Integer reasonCode;
+    @NotNull
+    @JsonPropertyDescription("A code uniquely identifying the cancellation reason")
+    private Integer reasonCode;
```

Comment:
The reason code is now in `Action` you can remove this.

Reply:
:+1: